File: src/main/kotlin/org/kite9/diagram/dom/model/LabelLeafImpl.java
package org.kite9.diagram.dom.model;

import org.kite9.diagram.batik.bridge.Kite9BridgeContext;
import org.kite9.diagram.dom.painter.LeafPainter;
import org.kite9.diagram.dom.elements.StyledKite9XMLElement;
import org.kite9.diagram.model.DiagramElement;
import org.kite9.diagram.model.Label;
import org.kite9.diagram.model.Leaf;
import org.kite9.diagram.model.style.ContentTransform;

/**
 * For text or shape-based labels within the diagram.
 * 
 * @author robmoffat
 * 
 */
public class LabelLeafImpl extends AbstractLabel implements Label, Leaf {
	
	public LabelLeafImpl(StyledKite9XMLElement el, DiagramElement parent, Kite9BridgeContext ctx, LeafPainter lo, ContentTransform t) {
		super(el, parent, ctx, lo, t);
	}
	
	
}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/dom/model/LabelLeafImpl.kt
package org.kite9.diagram.dom.model

import org.kite9.diagram.dom.bridge.ElementContext
import org.kite9.diagram.dom.painter.LeafPainter
import org.kite9.diagram.model.DiagramElement
import org.kite9.diagram.model.Label
import org.kite9.diagram.model.Leaf
import org.kite9.diagram.model.style.ContentTransform
import org.w3c.dom.Element

/**
 * For text or shape-based labels within the diagram.
 *
 * @author robmoffat
 */
class LabelLeafImpl(
    el: Element,
    parent: DiagramElement,
    ctx: ElementContext,
    lo: LeafPainter,
    t: ContentTransform
) : AbstractLabel(
    el, parent, ctx, lo, t
), Label, Leaf